pkg/controller/storage/bucketpolicymember.go

```diff
@@ -93,6 +95,9 @@ func (e *bucketPolicyMemberExternal) Observe(ctx context.Context, mg resource.Ma
 
 	instance, err := e.bucketpolicy.GetIamPolicy(gcp.StringValue(cr.Spec.ForProvider.Bucket)).OptionsRequestedPolicyVersion(iamv1alpha1.PolicyVersion).Context(ctx).Do()
 	if err != nil {
+		if strings.Contains(err.Error(), errNoSuchBucket) {
+			return managed.ExternalObservation{}, nil
+		}
 		return managed.ExternalObservation{}, errors.Wrap(err, errGetPolicy)
```

Comment:
I think you could use `Ignore` function to check if the resource was not found. I would suggest something like this:

```suggestion
		return managed.ExternalObservation{}, errors.Wrap(
			resource.Ignore(gcp.IsErrorNotFound, err),
			errGetPolicy,
		)
```